pkg/util/kubernetes/hpa/hpa.go

```diff
@@ -21,6 +21,11 @@ func Inspect(hpa *autoscalingv2.HorizontalPodAutoscaler) (emList []custommetrics
 	for _, metricSpec := range hpa.Spec.Metrics {
 		switch metricSpec.Type {
 		case autoscalingv2.ExternalMetricSourceType:
+			// The metricSelector and the external fields are optional. We do not support *.
+			if metricSpec.External == nil || metricSpec.External.MetricSelector == nil {
+				log.Errorf("Unexpected type for the HPA %#v, skipping processing", hpa)
```

Comment:
Seems kinda cryptic to me. Can we make the message more useful, or is it obvious in the context where it's displayed?

Reply:
made it clearer.